src/main/java/org/threeten/extra/chrono/InternationalFixedDate.java

```diff
@@ -0,0 +1,965 @@
+                    case ALIGNED_WEEK_OF_MONTH:
+                        return special ? InternationalFixedChronology.EMPTY_RANGE : ValueRange.of(1, InternationalFixedChronology.WEEKS_IN_MONTH);
+                    case ALIGNED_WEEK_OF_YEAR:
+                        return special ? InternationalFixedChronology.EMPTY_RANGE : InternationalFixedChronology.WEEK_OF_YEAR_RANGE;
+                    case DAY_OF_MONTH:
+                        return isYearDay() ? InternationalFixedChronology.EMPTY_RANGE
+                                           : isLeapDay() ? ValueRange.of(-1, -1)
+                                                         : ValueRange.of(1, InternationalFixedChronology.DAYS_IN_MONTH);
+                    case DAY_OF_YEAR:
+                        return isLeapYear() ? InternationalFixedChronology.DAY_OF_YEAR_LEAP_RANGE : InternationalFixedChronology.DAY_OF_YEAR_NORMAL_RANGE;
+                    case EPOCH_DAY:
+                        return InternationalFixedChronology.EPOCH_DAY_RANGE;
+                    case ERA:
+                        return InternationalFixedChronology.ERA_RANGE;
+                    case MONTH_OF_YEAR:
+                        return InternationalFixedChronology.MONTH_OF_YEAR_RANGE;
+                    default:
+                        break;
+                }
+            } else {
+                throw new UnsupportedTemporalTypeException("Unsupported field: " + field);
+            }
+        }
+
+        return super.range(field);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public int getAlignedDayOfWeekInMonth() {
+        if (day < 1) {
+            return 0;
+        }
+
+        return ((day - 1) % lengthOfWeek()) + 1;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    int getAlignedDayOfWeekInYear() {
+        if (day < 1) {
+            return 0;
+        }
+
+        return ((day - 1) % lengthOfWeek()) + 1;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    int getAlignedWeekOfMonth() {
+        if (day < 1) {
+            return 0;
+        }
+
+        return ((day - 1) / lengthOfWeek()) + 1;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    int getAlignedWeekOfYear() {
+        if (day < 1) {
+            return 0;
+        }
+
+        return 1 + (month - 1) * InternationalFixedChronology.WEEKS_IN_MONTH + ((day - 1) / InternationalFixedChronology.DAYS_IN_WEEK);
+    }
+
+    /**
+     * Returns the day of the week represented by this date.
+     * <p/>
+     * Leap Day and Year Day are not considered week-days, thus return 0.
+     *
+     * @return the day of the week: between 1 and 7, or 0 (Leap Day, Year Day)
+     */
+    @Override
+    public int getDayOfWeek() {
+        if (day < 1) {
+            return 0;
+        }
+
+        return 1 + ((day - 1) % InternationalFixedChronology.DAYS_IN_WEEK);
+    }
+
+    private int getCalculatedDayOfWeek() {
```

Comment:
`getCalculatedDayOfWeek()` should call `getDayOfWeek()` for normal days.  ('calculated' is also a terrible name, but I'm not coming up with anything good at the moment.  'equivalent', maybe?  If changed, don't forget month/day-of-month)